### ISSUE TITLE
Steam: Partially Fails to Connect to Internet after Waking

### ISSUE THREAD

**Issue by torval** [ValveSoftware/steam-for-linux#474]
I've noticed that, several times, after I have woken my computer up from "sleep," Steam will have difficulty connecting to the internet. Browsing and everything else on the PC works fine, but Steam will say that it is "Unable to connect to the server" when I attempt to load the store page.  Multiple refreshes and waiting a few minutes can usually resolve the issue.

Additionally, this gets in the way of launching games sometimes. For instance, right now, I am sitting waiting on Killing Floor to launch. The pop-up window says "Preparing to Launch Killing Floor," but it generally just sits there indefinitely, doing nothing. 

If I attempt to launch the game directly from it's working directory, it starts, but I receive a message about how multiplayer won't work, because Steam isn't running or I don't have an internet connection. 

Restarting Steam seems to resolve this.

**Comment by koraktor** [comment]
For me, this seems to be finally fixed after the [5/21 update](http://steamcommunity.com/groups/SteamClientBeta#announcements/detail/231152900027236983).

**Comment by mtmkls** [comment]
It seems to be fixed here, too.

**Comment by psychicteeth** [comment]
This issue persists for me on Mac OSX Yosemite with the latest Mac Steam client as of today.

**Comment by axfelix** [comment]
I still occasionally have this issue on both my Ubuntu (Unity) and Arch (Gnome) machines; Steam's network connection is occasionally only partially functional (won't sync cloud saves, downloads more slowly, etc.) after waking from suspend.

**Comment by herve-quiroz** [comment]
Same issue for me on Max OS Yosemite.

**Comment by licaon-kter** [comment]
Got it too, rather rarely now but a client restart fixes it.

**Comment by edalquist** [comment]
I see this regularly as well on 10.11 (and had the same behavior on 10.10, 10.9, etc)

**Comment by datkush420** [comment]
This has been an issue on Mac for as long as I can remember. It happens EVERY single time it wakes from sleep. Steam will say it's online, friends list won't update, no updates client/game come through, none of the pages are responsive; completely dead. Have to quit the application, which then goes unresponsive which then has to be force quit. After reopening, it runs perfect. That is, until the computer goes to sleep again. Then the above has to done again. This is on the latest OS X [non-beta] and the latest Steam client [non-beta]. I've had a mac for ~2yrs so this has been a DAILY ISSUE FOR 2 YEARS NOW. This issue does not happen on my Windows laptop or desktop. Only on Mac Mini/rMBP.

**Comment by DaveLessnau** [comment]
Seems to be happening with me, now.  Linux Mint 17.3 Cinnamon x64.  I'm not sure if it's related to the Linux Mint 17.2 to 17.3 update I did fairly recently or to a new version of Steam.

Steam client application
Built:  Dec 14 2015, at 11:19:30
Steam API: v017
Steam package versions:  1450127196

But, while this happened a few times in the past, it now happens EVERY time my system comes back from sleep.

**Comment by lracicot** [comment]
It also happens with me on mac. Every time I wake up my computer from sleep,  I have to close steam and reopen it, otherwise it won't connect to internet and won't lunch any game.

**Comment by andrew-hill** [comment]
Yep, I'm absolutely still seeing the issue on Mac OS X (18-months since my last post about it, everything up-to-date).

**Comment by darren-hayward** [comment]
I perplexed as to why it's been so long and Valve still has bothered to fix it. I mean you people are making millions off gamers and you can't fix a simple bug like this?

**Comment by jboyden** [comment]
I had this problem on my last computer, and now it's happening on my brand new computer so I'm no longer convinced that it was something to do with my computer. I've looked online for solutions previously and found nothing. This is the first time I saw this thread, it'd be great to have a solution.

**Comment by occivink** [comment]
Possibly related to the issue:
https://thesquareplanet.com/blog/the-story-of-a-rust-bug/

**Comment by sherief** [comment]
Issue still persists in 2017 on Windows 10. This is why competition is A Good Thing.

**Comment by JShorthouse** [comment]
Having the same issue on Xubuntu 16.04, steam fails to connect and joining any server without restarting steam gets me kicked after a few minutes for a VAC connection error.
Setting myself to offline and then back to online doesn't work as sometimes did on windows, I just remain offline permanently until I restart steam.

**Comment by alexandrebourlon** [comment]
Happens to me when one of my hard drive is sleeping (after being idle for a given time).
Part of my Steam library is on that disk, and starting Steam wakes it up. In this scenario Steam takes quite longer to start, and sometimes fails, but not always. What's weird after that is, if I launch a game (Dota 2), it starts, but has no connectivity whatsoever. Dota 2 is not even on that hard drive, it's on a separate SSD (main steam directory). I use the Steam Library Folders, no symlink/junction voodoo in Win10.

**Comment by masterxq** [comment]
I have the same problem and this should not be that hard to fix. This problem is really annoying. Why not fix it or at least set is as a milestone?
If no answer for some seconds test local connection to the internet by trying to reach a steam server, if success: the connection is broken and a reconnect is needed. Or just reduce the timeout... There are so many easy options...

**Comment by masterxq** [comment]
Another idea could be to have a service that periodically logs the time if there are more than x seconds gone from the last to the current sample, the computer just came from standby and the connection needs to be checked...

**Comment by tyler-hh** [comment]
I have this issue to.  Its pathetic.  Steam will auto update my games, I go to play them and while in game it says its not connected.  Even after 20 minutes of having a restored connection it still wont see it as being online.  This bug is 5 years old :(

**Comment by lewisdiamond** [comment]
Same issue, 6 years later. How about a fix boys?

**Comment by jonbitzen** [comment]
I have the same issue, and it makes the Steam Link experience much less polished than it could be.  

I have a few PCs and I set them to suspend with Steam Big Picture maximized when I'm not using one of the machines.  When I want to play a game on the TV, the Steam Link sends them a Wake On LAN packet which starts the computer up, and let's the Link reconnect to Steam on my PC.  

And it works great, except that my Steam Cloud connection on the PC goes away because of this network issue, which results in games taking forever to launch while the network stuff times out.

The Steam Link experience would be so much smoother with this fixed.

**Comment by JustJinxed** [comment]
2017, and the issue still exists. I thought at first it was because some of us had renamed the libstdc++.so shared object in order to get the client to work, (the rust bug suggests including a rewrite for res_init for hostlookups) but mine has reverted long ago, from updates. And hasn't needed to be removed for some time. 

So since it is now included fine, someone just needs to work out the kinks coming from an improper call to res_init for Ubuntu/Mint systems maybe? 

if it was any harder than exiting the client and making sure all the steamapi backgrounds close properly, and restarting... I might consider delving into the code myself. Think maybe you guys want to escalate this issue? It's been kinda a LONG haunt now.

**Comment by jonbitzen** [comment]
I'm more of a mind to hope that a Valve employee will read the comment about Steam Link, hope that they see it in their business interests to fix it to enhance the Steam Link experience, and prioritize the issue. :)

Not trying to get off on a tangent about Steam Link, but it is a pretty good experience (YMMV) except for this issue which is an irritant.

**Comment by ardje** [comment]
If read the comment about UDP, I guess it's just a firewall issue.
As udp is connectionless, it will never ever get teared down due to inconsistent ports or sequence id's.
But in the mean time the firewall might have a different opinion about said "connection" than the steam client.
What kind of router firewalls are involved? Has anyone tried to tcpdump the issue on the firewalls? Just to see what's going on...

**Comment by Selmi** [comment]
it is not only issue of lionux version. right now i am in windows 10, after sleep and wake up

```
Error
There was an error communicating with the Steam servers. Please try again later.
Retry
```
i didn't had this problem before, it started to appear only month or so ago. now i have it in both linux and windows system (on the same pc and same network)

**Comment by KKire** [comment]
This bug is from 2012, 6 YEARS AGO, and still not addressed. Also it doesn't reconnect when you switch WiFi networks. I was playing a game of dota 2 and I needed to switch to other WiFi network because the current stated lagging, so I switched network and reconnected to the game. a few minutes later, I'm kicked out of the game, because of 'Failed Steam Login'. So I needed to close dota 2, close steam and re-open it, start the game and then reconnect. This is so annoying and time-consuming.

Meanwhile on Windows, you can just click reconnect and done. Resuming from sleep or switching networks reconnect to Steam in ~10 seconds, while on Linux, Friends state ('Online', 'In-Game'...) gets stuck for ~1 hour before it reconnects and start being normal.

**Comment by kassybas** [comment]
The inaptitude of the Valve software engineers is baffling to me. How can you not debug and fix this.

**Comment by sherief** [comment]
@kassybas Because you can't answer "or else what?".
You're welcome.

**Comment by renatoliveira** [comment]
Installed Linux two days ago, and Steam yesterday. During the night I put the PC to sleep, and when I woke it up a couple of minutes ago I noticed this same issue. I've tried to download CS:GO, and got the timeout message two times before I lost the patience and just restarted Steam.

The thing is: this isn't the kind of issue that prevents someone from playing, but it is at least uncomfortable.

**Comment by jonbitzen** [comment]
I don't get upset with it as such.


It's just that if steam didn't have issues after waking up from sleep, I could have such a seamless Steam Link experience!  The Steam Link will send a wake-on-LAN package to your PC which is in sleep mode (I suspend to RAM).  And after it comes back up, I can play my games, but before they'll launch I have to wait through a network timeout while the game tries to talk to Steam Cloud to sync itself, but can't because of the network issue.


I have wired LAN in the house, and my Steam Link-PC experience is great.  I just wish I could have this additional convenience if the Steam Client would reconnect after sleep.


-



________________________________

**Comment by ashleybone** [comment]
Same years-old problem here.  Occasionally I have the same problem when resuming on Windows 7, but on Linux it happens almost every single time.

**Comment by dncrash** [comment]
I too am having this problem EVERY DAY. It's infuriating, especially since it's a known issue since 2012.

**Comment by mpsijm** [comment]
I am also experiencing this issue. Usually, about half an hour after waking up, Steam is able to reconnect. Too bad that this is still not fixed after all those years.

**Comment by volca02** [comment]
Every day issue for me on arch linux too. After I wake PC, I have to close steam and run it again.

**Comment by ashleybone** [comment]
I've noticed recently this is equally an issue on Windows 7.  I have to close steam and start it again, just like on Linux.

**Comment by riemers** [comment]
It could even be that the original poster died, since he has 6 years of no activity in his profile. (i moved back to windows for gaming only because of this issue)

**Comment by itcrowd72** [comment]
![screenshot from 2018-09-05 14-38-42](https://user-images.githubusercontent.com/29428813/45085203-93657a00-b119-11e8-8eb4-6b4c0a3fe223.png)

Same problem :(

**Comment by gueste** [comment]
I had the same issue today under the latest Windows 10 when resuming from hibernation.
I don't think that it happens everytime on Windows, but it's cumbersome, if you never can be sure if you are actually really "online" this time, or if it just seems to be that way.

Apart from the problem (occasionally) happening under Win 10, it's happening to me on Linux (Solus) every single time.

**Comment by af7567** [comment]
I am using Network Manager and I have always had this issue with the Linux client not reconnecting after system resume. I used to have the Metro skin installed which had a "Restart Steam" option in the menu which I would use to work around the problem.
The default Steam skin doesn't have a restart option though, so now I have to quit it and start it up again to get reconnected.

**Comment by howels** [comment]
Same bug, every time suspend/resume is used and on Windows as well as Linux. Games hang when launching because they try to synchronise save states with Steam Cloud. The app is not capable of reconnecting to the server on resume, which is kinda ridiculous as literally every other app does this.

**Comment by nymusicman** [comment]
See, this is weird because I've never had this issue before. But starting a week ago, all of a sudden it's happening. Running Antergos Linux Steam Runtime.

**Comment by dodwmd** [comment]
+1

**Comment by IvanDimitrov2002** [comment]
Same issue here running Manjaro Linux.

**Comment by jerheij** [comment]
Same issue here, Fedora 29. Steam doesn't reconnect properly after the computer wakes up from sleep.

**Comment by martadinata666** [comment]
Steam client application
Built: Jan 4 2019, at 23:23:48
Steam API: v018
Steam package versions: 1546909276

the weird issue is, after resume from sleep steam client can download automatically (steam can connect internet), but if i try to launch game it fail sync with steam data bla3, need to restart client to behave normal again.

**Comment by winny-** [comment]
Gentoo Linux same issue.

Steam client application
Built: Jon 30 2019, at 23:03:27
Steam API: v018
Steam package versions: 1548894698

[`emerge --info`, `eix-installed all`](https://gist.github.com/winny-/295c85a92c2d7dff155c8795d1ded5eb)

Making matters worse, whenever I accidentally hit "Remember Me" it crashes on first automatic login (next time one opens steam).

**Comment by sherief** [comment]
Doesn't reproduce on the Epic Games Store 😎

**Comment by jerheij** [comment]
> Doesn't reproduce on the Epic Games Store sunglasses
Unfortunately this issue is about Steam for Linux and **not** a platform that doesn't even support Linux yet. I don't see how that is even relevant in the slightest apart from trying to start some Epic Store vs Steam war.

**Comment by dodwmd** [comment]
Seems this issue has been open for 7 years with near zero feedback from the dev team it might as well have some humor in it.

**Comment by jerheij** [comment]
> Seems this issue has been open for 7 years with near zero feedback from the dev team it might as well have some humor in it.

I agree it's a "bit" sad nothing is being done, I guess I'm just tired of those Epic Store vs Steam wars... Or maybe it's just a bit too early and I'm turning old and cranky. ;)

**Comment by Wello6143** [comment]
Same with me here, like Valve devs can't fix this/too lazy to fix this annoying bug

**Comment by sinisab89** [comment]
Same bug on Linux Mint 19.1. Really nice job, Valve!

**Comment by pjf-valve** [comment]
G'day everyone!

Today's Steam Beta contains changes to the networking code that address this issue, although I suspect there may be multiple underlying causes, so we're hoping for feedback on how well it's working. **The Steam client can still take a few seconds to notice a network change**, but it should be well less than a minute.

So we don't overwhelm everyone on this ticket, if you find today's beta fixes things this bug for you, then please thumbs-up this post. If the issue still persists and its exactly the same as before, then please thumbs-down this post.

Many thanks!

~ Paul

**Comment by jerheij** [comment]
> Today's [Steam Beta](https://steamcommunity.com/groups/SteamClientBeta#announcements/detail/1611638091301723544) contains changes to the networking code that address this issue, although I suspect there may be multiple underlying causes, so we're hoping for feedback on how well it's working. **The Steam client can still take a few seconds to notice a network change**, but it should be well less than a minute.
> 
Even a minute is better than completely unable to reconnect, I am gonna attempt to update and test it!

**Comment by af7567** [comment]
I have tested a few times today after my PC had been to sleep and steam does seem to be reconnecting OK now within a minute and everything works properly after that.

I assume the minute is because steam is periodically checking something to see if it needs to reconnect? If steam accepted a signal which would force it to reconnect, or a command line argument, then a NetworkManager dispatcher script could be used which would tell steam the network has just reconnected. Obviously that would only work for people using NetworkManager, but the ability to force steam to reconnect by a command would be useful for other people too.

But yes, waiting a minute isn't that long :)

**Comment by pjf-valve** [comment]
> I assume the minute is because steam is periodically checking something to see if it needs to reconnect?

The time is a cautious balance between dropping connections too soon (which could result in flapping on a poor connection), waiting too long for a lost connection to return (which gives the pain point we've seen in this issue), and me not wanting to break Steam. :)

The current timing values are over 9000ms, but this isn't even their final form.

**Comment by pjf-valve** [comment]
Also noting the current Steam Beta may fix #971, #6183, and other issues caused by network connections taking a Very Long Time to realise there's something wrong.

**Comment by sinisab89** [comment]
It's working as it should now. Reconnects after a few seconds.

**Comment by FeralBytes** [comment]
Really?  I will have to test this... it has been so long I just always assumed it would fail.
Seems fixed like maybe we can finally close this one!

**Comment by dasunsrule32** [comment]
I'm on Ubuntu 19.04 and after waking from sleep, it almost always has issues with reconnecting to the internet. I have to exit steam, then open it again.

**Comment by howels** [comment]
Fixed for me in the recent betas. Ubuntu 19.04.

**Comment by dasunsrule32** [comment]
I'm on stable. Have the fixes for this issue been pushed to stable?

**Comment by pjf-valve** [comment]
@dasunsrule32 : It's only in the beta client for now, but since it seems to be working you can expect it to be folded in with the next stable release.

**Comment by dasunsrule32** [comment]
@pjf-valve : makes sense, thanks for the update!

**Comment by ivnish** [comment]
@pjf-valve 

From changelog 17th june 2019:

> Improved the client's responsiveness to network changes

Is this fix?

**Comment by ivanpu** [comment]
@ivnish looks like it, it works for me now in stable version (after few refreshes).

**Comment by dasunsrule32** [comment]
This is working for me in stable now. I have Steam package versions: 1560549607.

**Comment by ashleybone** [comment]
Fixed for me now.

**Comment by JustJinxed** [comment]
I haven't had a problem for awhile now, looks like this might have gotten some love <3

**Comment by kisak-valve** [comment]
Closing as fixed.

**Comment by riemers** [comment]
Should we make a honorable mention for this on https://developer.valvesoftware.com/wiki/Valve_Time ?

**Comment by cmgg** [comment]
> Should we make a honorable mention for this on https://developer.valvesoftware.com/wiki/Valve_Time ?

Sure, why not?

**Comment by laftur** [comment]
This issue is still very much alive on Arch Linux. After waking from suspend, no games will launch, indefinitely, until reboot.

**Comment by AnonFuncsAreAwesome** [comment]
Works for me on Archlinux.

**Comment by laftur** [comment]
> Works for me on Archlinux.

Reading the above comment history, it's difficult for me to determine what actually fixed this issue for the users here. Is it some kind of network management thing? My systems connected via ethernet can get away with just dhcpcd, but I digress. I feel like I'm making shots in the dark. For instance, the issue presents itself on both i3-wm and gnome.

Is there something about your system that you reckon might be affecting this issue? I guess we can rule out Arch Linux in general. I suspect the problem can be reproduced on other distros, but again, I have no idea how.

**Comment by mpsijm** [comment]
@laftur 

> it's difficult for me to determine what actually fixed this issue for the users here

A patch was released [(see this message)](https://github.com/ValveSoftware/steam-for-linux/issues/474#issuecomment-492855274) that fixed the network connectivity problems after waking from sleep.

> After waking from suspend, no games will launch, indefinitely, until reboot.

Do you also have this problem after restarting Steam? The original report (which was also my experience with this bug) stated that restarting Steam fixed the problem. If not, you probably have a different problem than this (already closed) issue, and I'd suggest opening a new issue. Hopefully you can find some more information about what is happening, to help pin down what the problem is.

**Comment by laftur** [comment]
> Do you also have this problem after restarting Steam? The original report (which was also my experience with this bug) stated that restarting Steam fixed the problem.

Restarting steam doesn't fix the issue. Even if I close steam before suspending, then start steam again after waking, no games will launch. I thought such an adjacent issue would have been investigated alongside this one. If you really think it's unrelated, I'll start another issue thread.
